src/discof/store/fd_storei_tile.c

```diff
@@ -205,6 +217,19 @@ during_frag( fd_store_tile_ctx_t * ctx,
     return;
   }
 
+  ctx->archive_publish=0;
+  if( FD_UNLIKELY( ctx->archive_out_mem ) ) {
+    uchar * dst = fd_chunk_to_laddr( ctx->archive_out_mem, ctx->archive_out_chunk );
+    if( in_idx==REPAIR_IN_IDX ) {
+      ctx->archive_publish=1;
+      fd_memcpy( dst, fd_chunk_to_laddr_const( ctx->repair_in_mem, chunk ), sz );
+    } else if( in_idx>=NON_SHRED_LINKS ) {
+      ctx->archive_publish=1;
+      fd_store_in_ctx_t * shred_in = &ctx->shred_in[ in_idx-NON_SHRED_LINKS ];
+      fd_memcpy( dst, fd_chunk_to_laddr_const( shred_in->mem, chunk ), sz );
+    }
+  }
```

Comment:
instead of adding conditional / copying here, do the copy into the archiver feeder out buffer in after_frag before you publish